### PR TITLE
Add prop to SocialPreviewEditor for replaceVars

### PR DESCRIPTION
**Pull request by bintzandt**
## Summary
Add prop to SocialPreviewEditor for replaceVars

This PR can be summarized in the following changelog entry:
* [@yoast/social-metadata-previews] Add a prop to the SocialPreviewEditor for replacement variables.

## Test instructions

This PR can be tested by following these steps:
See https://github.com/Yoast/wordpress-seo/pull/14930

## Quality assurance

* [x] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

### CODE REVIEW

**Comment by bintzandt** [comment]
The failing build is unrelated to this PR.

File: packages/social-metadata-previews/src/editor/SocialPreviewEditor.js [hunk 1]
```diff
@@ -128,10 +128,13 @@ class SocialPreviewEditor extends Component {
 			title,
 			replacementVariables,
 			recommendedReplacementVariables,
+			applyReplacementVariables,
 			isPremium,
 			isLarge,
 		} = this.props;
 
+		const replacedVars = applyReplacementVariables( { title, description } );
```
Comment (maartenleenders) [inline packages/social-metadata-previews/src/editor/SocialPreviewEditor.js:136]: `applyReplacementVariables` expects `data.url` to exist, which is the args you pass here.

Either change the function to have a fallback for data.url, pass an empty url here, or let it be?

Comment (maartenleenders) [reply packages/social-metadata-previews/src/editor/SocialPreviewEditor.js:136]: We can live with it being undefined for now

**Comment by maartenleenders** [comment]
> The failing build is unrelated to this PR.

Btw, I cannot merge with failing builds, unrelated or not 😉

**Pull request merged**